src/lib/containers/EntityBadgeIcons.tsx

```diff
@@ -189,121 +196,121 @@ export const EntityBadgeIcons: React.FunctionComponent<EntityBadgeIconsProps> =
           : ''
       }`
 
-  if (bundle == null) {
-    return null
-  }
-
   return (
-    <div className="EntityBadge" style={{ flexWrap, justifyContent }}>
-      <ReactTooltip
-        id={ENTITY_BADGE_ICONS_TOOLTIP_ID}
-        className="EntityBadgeTooltip"
-        delayShow={100}
-        place={'right'}
-      />
-      <div onClick={e => e.stopPropagation()}>
-        <EntityModal
-          entityId={entityId}
-          show={showModal}
-          showTabs={false}
-          onClose={() => setShowModal(false)}
-          initialTab={EntityModalTabs.ANNOTATIONS}
-        />
-      </div>
-      {showIsPublicPrivate && bundle.benefactorAcl && isPublic(bundle) ? (
-        <FontAwesomeIcon
-          aria-hidden={false}
-          role="img"
-          className="EntityBadge__Badge"
-          icon={faGlobeAmericas}
-          data-for={ENTITY_BADGE_ICONS_TOOLTIP_ID}
-          data-tip={'Public'}
-          data-testid={'is-public-icon'}
-        />
-      ) : null}
-      {showIsPublicPrivate && bundle.benefactorAcl && !isPublic(bundle) ? (
-        <FontAwesomeIcon
-          aria-hidden={false}
-          role="img"
-          className="EntityBadge__Badge"
-          icon={faLock}
-          data-for={ENTITY_BADGE_ICONS_TOOLTIP_ID}
-          data-tip={'Private'}
-          data-testid={'is-private-icon'}
-        />
-      ) : null}
-      {showHasLocalSharingSettings &&
-      bundle.benefactorAcl &&
-      entityId === bundle.benefactorAcl!.id ? (
-        <FontAwesomeIcon
-          aria-hidden={false}
-          role="img"
-          className="EntityBadge__Badge"
-          icon={faCheck}
-          data-for={ENTITY_BADGE_ICONS_TOOLTIP_ID}
-          data-tip="Sharing Settings have been set"
-          data-testid={'sharing-settings-icon'}
-        />
-      ) : null}
-
-      {showHasAnnotations &&
-        !!(annotationsCount || schemaValidationResults) && (
-          <FontAwesomeIcon
-            aria-hidden={false}
-            role={canOpenModal ? 'button' : 'img'}
-            className={`EntityBadge__Badge ${schemaConformance}`}
-            style={canOpenModal ? { cursor: 'pointer' } : undefined}
-            onClick={canOpenModal ? () => setShowModal(true) : undefined}
-            icon={faTag}
-            data-for={ENTITY_BADGE_ICONS_TOOLTIP_ID}
-            data-tip={sanitizeHtml(annotationsHtml)}
-            data-html={true}
-            data-testid={'annotations-icon'}
+    <div ref={ref} className="EntityBadge" style={{ flexWrap, justifyContent }}>
+      {bundle && (
```

Comment:
Moved this check inside the `div` so we can attach the ref to check to see if the component comes into view.